File: keevo/app/lib/features/inventory/domain/model/inventory_scope.dart
/// InventoryScope — scope of an inventory counting session.
/// Story 6.1.
enum InventoryScope {
  full,
  partial;

  String toJson() => name.toUpperCase();

  static InventoryScope fromJson(String value) {
    switch (value.toUpperCase()) {
      case 'FULL':
        return InventoryScope.full;
      case 'PARTIAL':
        return InventoryScope.partial;
      default:
        throw ArgumentError('Unknown InventoryScope: $value');
    }
  }
}

File: keevo/backend/src/main/java/com/keevo/inventory/counting/domain/model/InventoryScope.java
package com.keevo.inventory.counting.domain.model;

/**
 * InventoryScope — Defines the scope of an inventory session.
 *
 * <p>FULL = count all active products in the store.
 * PARTIAL = count only products in specific categories.
 */
public enum InventoryScope {
    FULL,
    PARTIAL
}
